public/app/features/query/state/runRequest.ts

```diff
@@ -195,6 +201,15 @@ export function callQueryMethod(
       : t
   );
 
+  // do not filter queries in case a custom query function is provided (for example in variable queries)
+  if (!queryFunction) {
+    request.targets = request.targets.filter((t) => datasource.filterQuery?.(t) ?? true);
```

Comment:
this confuses me a bit. Doesn't this mean that the target is filtered _before_ being executed?

Reply:
sorry, I got confused with hidden / filtered :upside_down_face: